builds/e2e/templates/e2e-run.yaml

```diff
@@ -90,7 +90,7 @@ steps:
     Copy-Item "$(binDir)/*-device-*.log" "$logDir/"
     Copy-Item "$(binDir)/testoutput.log" "$logDir/"
     Copy-Item "$(binDir)/supportbundle*" "$logDir/"
-    $artifactSuffix = '$(Build.BuildNumber)-$(System.PhaseName)' -replace '_','-'
+    $artifactSuffix = '$(Build.BuildNumber)-$(System.JobName)' -replace '_','-'
```

Comment:
What's the difference between PhaseName and JobName?

Reply:
Also, your 1.2 PR doesn't make this change??

Reply:
I made this change to fix an issue causing the artifacts for all matrix elements to collide, resulting in one artifact for the last scenario to complete, rather than one for each scenario. In this case, I found that PhaseName resolves to 'partial-upgrade-ubuntu-1804-msmoby' but JobName resolves to the label used for the matrix element (or 'Job<n>' when the labels are too long).

Good catch on the diff with the 1.2 PR. I need to make the same change there because the artifacts are colliding.

I'll also confirm that this doesn't negatively affect the artifacts for regular E2E pipeline runs.